packages/sdks/e2e/angular-ssr/src/app/app.component.ts

```diff
@@ -48,31 +54,34 @@ export class AppComponent {
     },
   ];
 
-  constructor(private cdr: ChangeDetectorRef) {}
+  constructor(
+    private cdr: ChangeDetectorRef,
+    @Inject(PLATFORM_ID) private platformId: any
+  ) {}
 
   async ngOnInit() {
-    const urlPath = window.location.pathname || '';
-
-    const builderProps = await getProps({
-      pathname: urlPath,
-      _processContentResult,
-      options: getBuilderSearchParams(
-        new URLSearchParams(window.location.search)
-      ),
-      fetchOneEntry,
-    });
+    if (isPlatformBrowser(this.platformId)) {
```

Comment:
I see the issue now. This SSR example was never fully working on the server, since it relied on `window`. By adding this browser check here, you're essentially disabling SSR: the `builderProps` are only being set once we have access to the `window`.

We need a way to access this information on the server. Here's what I found from looking around:

- this shows a way to do it in an angular service: https://stackoverflow.com/questions/78417872/how-to-get-query-params-with-angular-17-ssr-on-server-side
- I see this bit about injecting the Request token https://www.npmjs.com/package/@nguniversal/express-engine#using-the-request-and-response

Reply:
you're right!! my bad. I realised it late trying to figure out the `window` issue, let me try these! Thanks!!